Prevent duplicates in groovy.log appenders

marcpa00 opened the pull request:
Do not add a new ConsoleAppender to groovy.log logger if one is already present.

Without this patch, when a ConsoleAppender have been already defined in the soapui-log4j.xml file, initGroovyLog() will add a second one.  

Also, without this patch, when called from SoapUITestCaseRunner (through soapui-maven-plugin for example), every invocation will add another appender, because a new instance of SoapUITestCaseRunner is created (thus groovyLogInitialized is false) but the logger we get from Logger.getLogger("groovy.log") is the instance already configured in the JVM; we get as many ConsoleAppender instances as SoapUITestCaseRunner.run() invocations.

I think this is related to  SOAP-1938, but I'm not sure because I don't have access to your jira issues.

marcpa00 commented:
Github seems to think the whole file has changed, must be a line-ending problem... on my side, git reports : 

 1 file changed, 15 insertions(+), 4 deletions(-)

Anyway, let me know if you want me to create another commit.

redfish4ktc commented:
Hi, for maven plugin users, this is already implemented in maven-soapui-extension-plugin, see redfish4ktc/maven-soapui-extension-plugin#68

spindelmanne commented:
@marcpa00 If you could create another commit, that'd be great.

spindelmanne commented:
Friendly reminder, @marcpa00 : could you fix the line endings to make my work easier? :-)

marcpa00 commented:
Oh, I see, end-of-line nightmare !  Sorry about that.

I've reworked the fix into another branch, making sure the patched file stays with its original end-of-line settings and I added a test project in soapui-maven-plugin-tester.
I'll create another pull request for this.

PS: I think that adding `* text=auto` at start of `.gitattributes` in top-level directory would prevent this kind of problems, if I understand correctly what is said here : http://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/

marcpa00 commented:
Closing this pull request as it is superceeded by #198
